lib/Sema/TypeCheckStorage.cpp

```diff
@@ -701,22 +737,42 @@ static Expr *buildStorageReference(AccessorDecl *accessor,
       firstWrapperIdx = 1;
 
     // Perform accesses to the wrappedValues along the composition chain.
-    for (unsigned i : range(firstWrapperIdx, lastWrapperIdx)) {
-      auto wrapperInfo = var->getAttachedPropertyWrapperTypeInfo(i);
-      auto wrappedValue = wrapperInfo.valueVar;
-
-      // Check for availability of wrappedValue.
-      if (accessor->getAccessorKind() == AccessorKind::Get ||
-          accessor->getAccessorKind() == AccessorKind::Read) {
-        if (wrappedValue->getAttrs().getUnavailable(ctx)) {
-          diagnoseExplicitUnavailability(
-              wrappedValue,
-              var->getAttachedPropertyWrappers()[i]->getRangeWithAt(),
-              var->getDeclContext(), nullptr);
+    if (firstWrapperIdx < lastWrapperIdx) {
+      if (auto lvalueness = getPropertyWrapperLValueness(var)) {
```

Comment:
`firstWrapperIdx < lastWrapperIdx` guarantees that `getPropertyWrapperLValueness` will return a value, so this `if` can be replaced with

```
auto lvalueness = *getPropertyWrapperLValueness(var);
```

right?